android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BroadcastActivity.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2014 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+BroadcastActivity.java
+
+This is the main entry point for the application
+It is the primary ui interface
+It also creates the singletons used for the main app logic
+like sharing a URL via bluetooth, and interfacing with Drive
+It also contains the logic for ensuring permissions
+and other account requirements are satisfied
+*/
+
+package org.physical_web.physicalweb;
+
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.bluetooth.BluetoothAdapter;
+import android.bluetooth.BluetoothManager;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Build;
+import android.util.Log;
+import android.util.Patterns;
+import android.widget.Toast;
+
+import org.physical_web.physicalweb.ble.UriBeacon;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+
+
+/**
+ * Handles share intent.
+ */
+public class BroadcastActivity extends Activity {
+
+    private static final String TAG  = "BroadcastActivity";
+    private static final int ENABLE_BLUETOOTH_REQUEST_ID = 1;
+    public static final int MAX_URI_LENGTH = 18;
+    private static final String SHARE_CONST = "URL";
+    private static final String DISP_CONST = "OURL";
+    private String url;
+
+
+    // callbacks
```

Comment:
You lost the first half of this formatted comment...I'm fine with you just removing the whole thing.